File: Insertion-Sort---Part-2.cpp
void printArray(const vector<int>& arr) {
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;
}

void insertionSort2(int n, vector<int>& arr) {
    for (int i = 1; i < n; ++i) {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;

        printArray(arr);
    }
}

template <typename T>
void show_value(const T& arr)
{
    for (auto element : arr)
    {
        cout << element << ' ';
    }
    
    cout << '\n';
}

void insertionSort2(int n, vector<int> arr)
{
    for (int i = 1; i < n; ++i)
    {
        int key = arr[i];
        int j;

        for (j = i-1; j >= 0 && arr[j] > key; --j)
        {
            arr[j+1] = arr[j];
        }

        arr[j+1] = key;
        show_value(arr);
    }
}

File: Insertion-Sort---Part-2.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'insertionSort2' function below.
     *
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER_ARRAY arr
     */

    static void showValue(List<Integer> arr) {
        for (int num : arr) System.out.print(num + " ");
        System.out.println();
    }

    public static void insertionSort2(int n, List<Integer> arr) {
        for (int i = 1; i < n; ++i) {
            int key = arr.get(i);
            int j = i - 1;
            while (j >= 0 && arr.get(j) > key) {
                arr.set(j + 1, arr.get(j));
                --j;
            }
            arr.set(j + 1, key);
            showValue(arr);
        }
    }    
}
